sound/soc/sof/ops.h

```diff
@@ -243,8 +243,10 @@ snd_sof_dsp_set_power_state(struct snd_sof_dev *sdev,
 /* debug */
 static inline void snd_sof_dsp_dbg_dump(struct snd_sof_dev *sdev, u32 flags)
 {
-	if (sof_ops(sdev)->dbg_dump)
+	if (sof_ops(sdev)->dbg_dump && !sdev->dbg_dump_printed) {
 		sof_ops(sdev)->dbg_dump(sdev, flags);
+		sdev->dbg_dump_printed = true;
+	}
```

Comment:
Hmm, not quite comfortable with this. The dbg_dump() can be used on non-error flows as well. Should this rate-limiting be done only for the error cases?

Also, with runtime-PM, we do have basic form of recovery:
   - DSP panics, use gets an error pop up in application
   - application closed
   - ALSA PCM closed
   - SOF goes to runtime suspsend
   - user reopens the app
   - ALSA PCM opened again
   - new DSP boot and we typically do recover

We would not want to miss any follow-panics that happen after the above.

So any ratelimiting of DSP dumps should be done in context of a single DSP boot and we should reset state if we boot again.

Reply:
I see, we then should reset the `sdev->dbg_dump_printed` after successful DSP boots.
Let me check that.

Reply:
Setting the  `sdev->dbg_dump_printed` to `false` in `snd_sof_run_firmware()` will re-enable the dump after each DSP boot.
I'm not certain how platforms with 'support for low power D0 substate' is handled, how they are rebooted if they are at all?

Reply:
@ujfalusi This should be ok and should cover also D0 substate cases sufficiently. If DSP panics, the low-power pipelines will return an error as well. If the device is not let go to runtime suspend after an error, the problem will then persist, but is no longer an issue with the debug dumps, but rather how to recover from these situations. And basicly for successful recovery, one has to reboot the DSP again and that will mean  a call to snd_sof_run_firmware(). As long as any future DSP recovery mechanism remembers to clean the dbg_dump_printed flag (via calling snd_sof_run_firwmare() or directly), we should be good.